### PR TITLE
[SPARK-28805][DOCS][SQL] Document DESCRIBE FUNCTION in SQL Reference

### PR DESCRIPTION
**Pull request by dilipbiswal**
### What changes were proposed in this pull request?
Document DESCRIBE FUNCTION statement in SQL Reference Guide. 

### Why are the changes needed?
Currently Spark lacks documentation on the supported SQL constructs causing
confusion among users who sometimes have to look at the code to understand the
usage. This is aimed at addressing this issue.

### Does this PR introduce any user-facing change?
Yes. 

**Before:**
There was no documentation for this.

**After.**
Screen Shot 2019-09-02 at 11 14 09 PM
Screen Shot 2019-09-02 at 11 14 29 PM
Screen Shot 2019-09-02 at 11 14 42 PM

### How was this patch tested?
Tested using jykyll build --serve

### CODE REVIEW

**Comment by dilipbiswal** [comment]
cc @gatorsmile

File: docs/sql-ref-syntax-aux-describe-function.md [hunk 1]
```diff
@@ -18,5 +18,40 @@ license: |
   See the License for the specific language governing permissions and
   limitations under the License.
 ---
+### Description
 
-**This page is under construction**
+Return the metadata such as qualified function name, implementing class and usage of an existing function.
+Input parameter function name may be optionally qualified. If the function is not qualified then the function is
+resolved from the current database. If EXTENDED option is specified then additionally extended usage information
```
Comment (srowen) [inline docs/sql-ref-syntax-aux-describe-function.md]: Back-tick `EXTENDED`, and add "the" before.

**Review by srowen — CHANGES_REQUESTED** [review]

File: docs/sql-ref-syntax-aux-describe-function.md (hunk 1 above)
Comment (dilipbiswal) [reply docs/sql-ref-syntax-aux-describe-function.md]: @srowen Changed. Sean, i also backticked the function_name. Is that ok ?

**Comment by gatorsmile** [comment]
LGTM

Thanks! Merged to master.

**Comment by dilipbiswal** [comment]
@gatorsmile Thank you very much for fixing the mistakes and merging.

**Pull request closed without merging**